### PR TITLE
HPCC-30784 Support getCursor() and setCursor() in ParquetRowReader

### PR DESCRIPTION
**Pull request by jackdelv**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by dcamper** [comment]
@jackdelv The changes here seem to be far-ranging, beyond just what is described in the commit message.  Is it possible that your specific changes should be merged after another PR is merged?

**Comment by jackdelv** [comment]
@dcamper Yes, this branch is based off the branch in this [PR](https://github.com/hpcc-systems/HPCC-Platform/pull/18233). Only the final commit pertains to this PR. I can rebase to the upstream master if it is an issue.

**Comment by dcamper** [comment]
Got it.  And now I see that this is a draft anyway....

**Review by dcamper — APPROVED** [review]
Reviewed only the latest commit

File: plugins/parquet/parquetembed.cpp [hunk 1]
```diff
@@ -406,6 +410,70 @@ __int64 ParquetReader::next(TableColumns *&nextTable)
...
+void ParquetReader::setCursor(MemoryBuffer & cursor)
+{
+    restoredCursor = true;
+    tablesProcessed = 0;
+    totalRowsProcessed = 0;
+    rowsProcessed = 0;
+    rowsCount = 0;
+
+    bool partition;
+    cursor.read(partition);
```
Comment (ghalliday) [inline plugins/parquet/parquetembed.cpp:463]: You could sanity check that partition matches endsWithIgnoreCase(partOption.c_str(), "partition");

**Review by ghalliday — APPROVED** [review]
It all looks sensible. although I can't tell if it works...

File: plugins/parquet/parquetembed.cpp (hunk 1 above)
Comment (jackdelv) [reply plugins/parquet/parquetembed.cpp:463]: I think that is a good check to have. Should I create a seperate JIRA for it, or wait for changes from another subtask?

I did test the setCursor/getCursor by initializing a new ParquetReader and stepping backwards through a random section of the table.

**Pull request merged**